### PR TITLE
feat: networking: add healthz and livez endpoints

### PR DESCRIPTION
**Pull request by coryschwartz**
## Proposed Changes

* New endpoints healthz and livez, this is helpful for kubernetes readiness and liveness probes.

* not included in this change, we will want to add an additional readiness probe to the gateway

## Additional Info
see https://kubernetes.io/docs/reference/using-api/health-checks/
On k8s, health endpoints like this are used to move nodes into and out of service and load balancer pools.

## Checklist

Before you mark the PR ready for review, please make sure that:
- [x] All commits have a clear commit message.
- [x] The PR title is in the form of of `<PR type>: <area>: <change being made>`
    - example: ` fix: mempool: Introduce a cache for valid signatures`
    - `PR type`: _fix_, _feat_, _INTERFACE BREAKING CHANGE_, _CONSENSUS BREAKING_, _build_, _chore_, _ci_, _docs_,_perf_, _refactor_, _revert_, _style_, _test_
    - `area`: _api_, _chain_, _state_, _vm_, _data transfer_, _market_, _mempool_, _message_, _block production_, _multisig_, _networking_, _paychan_, _proving_, _sealing_, _wallet_, _deps_
- [ ] This PR has tests for new functionality or change in behaviour
- [ ] If new user-facing features are introduced, clear usage guidelines and / or documentation updates should be included in https://lotus.filecoin.io or [Discussion Tutorials.](https://github.com/filecoin-project/lotus/discussions/categories/tutorials)
- [x] CI is green

### CODE REVIEW

File: node/health.go [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+	ctx := context.Background()
+	h := HealthHandler{}
+	go func() {
+		const reset = 5
+		var countdown = 0
+		minutely := time.NewTicker(time.Minute)
+		headCh, err := api.ChainNotify(ctx)
+		if err != nil {
+			//TODO
+		}
```
Comment (magik6k) [inline node/health.go:40]: Could call outside of the goroutine and return the error.

File: node/health.go [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+type HealthHandler struct {
+	healthy bool
+}
+
+func (h *HealthHandler) SetHealthy(healthy bool) {
+	h.healthy = healthy
+}
+
+func (h *HealthHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
+	if !h.healthy {
```
Comment (magik6k) [inline node/health.go:21]: This isn't thread-safe, slightly better version which will make the race-detector not complain:
```suggestion
	healthy int32
}

func (h *HealthHandler) SetHealthy(healthy bool) {
	h := int32(0)
	if healthy {
		h = 1
	}
	atomic.StoreInt32(&h.healthy, h)
}

func (h *HealthHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
	if atomic.LoadInt32(&h.healthy) != 1 {
```

File: node/health.go [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+				h.SetHealthy(true)
+			}
+		}
+	}()
+	return &h
+}
+
+// Check if we are ready to handle traffic.
+// 1. sync workers are caught up.
+// 2
```
Comment (magik6k) [inline node/health.go:59]: Missing

File: node/health.go [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+			return
+		}
+		for {
+			select {
+			case <-minutely.C:
+				atomic.AddInt32(&countdown, -1)
+				if countdown <= 0 {
+					h.SetHealthy(false)
+				}
+			case <-headCh:
```
[the remark this answers is not in the document]
Comment (coryschwartz) [reply node/health.go:58]: Good point. Thank you.

**Review by magik6k — APPROVED** [review]

**Pull request merged**